cloudcix_primitives/nat_firewall_ns.py

```diff
@@ -1,18 +1,305 @@
+            return False, fmt.payload_error(ret, f"{prefix+2}: " + messages[prefix+2]), fmt.successful_payloads
+        fmt.add_successful('flush_postrouting', ret)
+
+        ret = rcc.run(payloads['flush_prerouting'])
+        if ret["channel_code"] != CHANNEL_SUCCESS:
+            return False, fmt.channel_error(ret, f"{prefix+3}: " + messages[prefix+3]), fmt.successful_payloads
+        if ret["payload_code"] != SUCCESS_CODE:
+            return False, fmt.payload_error(ret, f"{prefix+4}: " + messages[prefix+4]), fmt.successful_payloads
+        fmt.add_successful('flush_prerouting', ret)
+
+        for mapping in one_to_one:
+            payload_prerouting = rule_templates['prerouting_11'] % mapping
+            payload_postrouting = rule_templates['postrouting_11'] % mapping
+
+            ret = rcc.run(payload_postrouting)
+            if ret["channel_code"] != CHANNEL_SUCCESS:
+                return False, fmt.channel_error(ret, f"{prefix+5}: " + messages[prefix+5] % {'payload': payload_postrouting}), fmt.successful_payloads
+            if ret["payload_code"] != SUCCESS_CODE:
+                return False, fmt.payload_error(ret, f"{prefix+6}: " + messages[prefix+6] % {'payload': payload_postrouting}), fmt.successful_payloads
+            fmt.add_successful('postrouting_11 (%s)' % payload_postrouting, ret)
+
+            ret = rcc.run(payload_prerouting)
+            if ret["channel_code"] != CHANNEL_SUCCESS:
+                return False, fmt.channel_error(ret, f"{prefix+7}: " + messages[prefix+7] % {'payload': payload_prerouting}), fmt.successful_payloads
+            if ret["payload_code"] != SUCCESS_CODE:
+                return False, fmt.payload_error(ret, f"{prefix+8}: " + messages[prefix+8] % {'payload': payload_prerouting}), fmt.successful_payloads
+            fmt.add_successful('prerouting_11 (%s)' % payload_prerouting, ret)
+
+        for network in ranges:
+            payload = rule_templates['range'] % {'network': network}
+
+            ret = rcc.run(payload)
+            if ret["channel_code"] != CHANNEL_SUCCESS:
+                return False, fmt.channel_error(ret, f"{prefix+9}: " + messages[prefix+9] % {'payload': payload}), fmt.successful_payloads
+            if ret["payload_code"] != SUCCESS_CODE:
+                return False, fmt.payload_error(ret, f"{prefix+10}: " + messages[prefix+10] % {'payload': payload}), fmt.successful_payloads
+            fmt.add_successful('range (%s)' % payload, ret)
+
+        return True, "", fmt.successful_payloads
+
+
+    status, msg, successful_payloads = run_podnet(enabled, 3020, {})
+    if status == False:
+        return status, msg
+
+    status, msg, successful_payloads = run_podnet(disabled, 3060, successful_payloads)
+    if status == False:
+        return status, msg
+
+    return True, messages[1000]
 
 
 def read() -> Tuple[bool, dict, str]:
-    return(False, {}, 'Not Implemted')
+    return(False, {}, 'Not Implemented')
+
+
+def scrub(
```

Comment:
There is no need for a scrub method for the NAT table PREROUTING and POSTROUTING chains. If it is required to remove all NATs, it can be managed via the build request by sending an empty list for `one_to_one` and `ranges`.

When a Project namespace is to be deleted from the PodNet boxes the scrubbing of the user chains will be managed by calling the `default_firewall_ns.scrub()` primitive. 

The same reasoning was used for not requiring scrub in the other user chain issues.

Reply:
Ok, I removed the scrub() method now.